test/RelayHub.test.ts

```diff
@@ -0,0 +1,1460 @@
+            forwarder,
+            relayRequestMisbehavingVerifier,
+            owner
+          );
+
+          const relayCall = relayHub
+            .connect(relayWorker)
+            .relayCall(relayRequestMisbehavingVerifier, signature, {
+              gasLimit,
+              gasPrice,
+            });
+
+          await expect(relayCall).to.be.rejectedWith('Invalid gas price');
+        });
+
+        it('should accept relay requests with incorrect relay worker', async function () {
+          await relayHub
+            .connect(relayManager)
+            .addRelayWorkers([incorrectWorker.address]);
+          await relayHub
+            .connect(relayWorker)
+            .relayCall(
+              relayRequestMisbehavingVerifier,
+              signatureWithMisbehavingVerifier,
+              {
+                gasLimit,
+                gasPrice,
+              }
+            );
+
+          const transactionRelayedRevertedFilter =
+            relayHub.filters.TransactionRelayedButRevertedByRecipient();
+          const transactionRelayedRevertedEvent = await relayHub.queryFilter(
+            transactionRelayedRevertedFilter
+          );
+
+          expect(transactionRelayedRevertedEvent.length).to.equal(1);
+        });
+      });
+    });
+  });
+
+  describe('deployCall', function () {
+    const minIndex = 0;
+    const maxIndex = 1000000000;
+    let nextWalletIndex: number;
+    let deployRequest: DeployRequest;
+    let deployVerifier: TestDeployVerifierEverythingAccepted;
+    let unknownWorker: SignerWithAddress;
+
+    beforeEach(async function () {
+      nextWalletIndex = Math.floor(
+        Math.random() * (maxIndex - minIndex + 1) + minIndex
+      );
+
+      deployVerifier = await deployContract(
+        'TestDeployVerifierEverythingAccepted'
+      );
+      [unknownWorker] = (await ethers.getSigners()) as [SignerWithAddress];
+
+      deployRequest = {
+        request: {
+          relayHub: relayHub.address,
+          to: ethers.constants.AddressZero,
+          data: '0x',
+          from: owner.address,
+          nonce: (await factory.nonce(owner.address)).toString(),
+          value: '0',
+          tokenContract: token.address,
+          tokenAmount: '1',
+          tokenGas: '50000',
+          recoverer: ethers.constants.AddressZero,
+          index: '0',
+          validUntilTime: '0',
+        },
+        relayData: {
+          gasPrice,
+          feesReceiver: relayWorker.address,
+          callForwarder: factory.address,
+          callVerifier: deployVerifier.address,
+        },
+      };
+    });
+
+    const cloneDeployRequest = (
+      override: Partial<{
+        request: Record<string, unknown>;
+        relayData: Record<string, unknown>;
+      }>
+    ) => {
+      return {
+        request: { ...deployRequest.request, ...override.request },
+        relayData: { ...deployRequest.relayData, ...override.relayData },
+      };
+    };
```

Comment:
This function looks similar to `cloneRelayRequest`

Reply:
I have rearranged it to avoid code duplication, but both cloneDeployRequest and cloneRelayRequest are still needed for type reasons. Unless there's something I'm not seeing.